File: 1089. Duplicate Zeros/DuplicateZeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        shift = arr.count(0)
        
        for i in range(len(arr)-1,-1,-1):

            if arr[i] == 0:
                shift -= 1
                if i+shift+1 < len(arr):
                    arr[i+shift+1] = 0

            if i+shift < len(arr):
                arr[i+shift] = arr[i]

File: 1089. Duplicate Zeros/DuplicateZeros.cs
public class Solution {
    public void DuplicateZeros(int[] arr) {
        
        int shift = arr.Count(x => x == 0);
        
        for (int i = arr.Length-1; i >= 0; --i) {

            if (arr[i] == 0) {

                shift--;
                //duplicate the zero
                if (i+shift+1 < arr.Length) {
                    arr[i+shift+1] = 0;
                }
            }

            if (i+shift < arr.Length) {
                arr[i+shift] = arr[i];
            }
        }
    }
}
